src/proposal/mod.rs

```diff
@@ -362,7 +367,14 @@ impl Contract {
             self.label_to_proposals.insert(&label_to_add, &proposals);
         }
 
+        self.add_change_log(ChangeLog {
+            block_id: env::block_height(),
+            block_timestamp: env::block_timestamp(),
+            changed_object_id: id,
+            change_log_type: ChangeLogType::Proposal(id),
+        });
```

Comment:
Since block_height and block_timestamp are always used and if we can get rid of `changed_object_id`, I would suggest to change the API to only require the ChangeLogType to be passed:

```suggestion
        self.add_change_log(ChangeLogType::Proposal(id));
```

Reply:
Good point